### PR TITLE
[zh-tw] create the translation of `Object.entries()`

### PR DESCRIPTION
**Pull request by BlueBoy247**
### Description

I created a new page to translate [this page](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries) into Traditional Chinese.

### Motivation

The document I translated before has a link to this document, and found that it hasn't been translated, so I want to translate it as well, hoping to help more readers.

### Additional details

No.

### Related issues and pull requests

No.

### CODE REVIEW

File: files/zh-tw/web/javascript/reference/global_objects/object/entries/index.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
+---
+title: Object.entries()
+slug: Web/JavaScript/Reference/Global_Objects/Object/entries
+---
```
Comment (T34-active) [inline files/zh-tw/web/javascript/reference/global_objects/object/entries/index.md:4]:
```suggestion
---
title: Object.entries()
slug: Web/JavaScript/Reference/Global_Objects/Object/entries
l10n:
  sourceCommit: 4ce6b9526bfa5b44a518e8ecb21a9894973136bd
---
```

File: files/zh-tw/web/javascript/reference/global_objects/object/entries/index.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+{{Specifications}}
+
+## 瀏覽器相容性
+
+{{Compat}}
+
+## 參見
+
+- [Polyfill of `Object.entries` in `core-js`](https://github.com/zloirock/core-js#ecmascript-object)
```
Comment (T34-active) [inline files/zh-tw/web/javascript/reference/global_objects/object/entries/index.md:111]:
```suggestion
- [`core-js` 中 `Object.entries` 的 Polyfill](https://github.com/zloirock/core-js#ecmascript-object)
```

**Review by T34-active — COMMENTED** [review]
別忘了寫 l10n

File: files/zh-tw/web/javascript/reference/global_objects/object/entries/index.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
+---
+title: Object.entries()
+slug: Web/JavaScript/Reference/Global_Objects/Object/entries
+l10n:
+  sourceCommit: 4ce6b9526bfa5b44a518e8ecb21a9894973136bd
+---
+
+{{JSRef}}
+
+**`Object.entries()`** 靜態方法回傳給定之物件自己的可枚舉字串鍵控屬性的鍵值對的陣列。
```
Comment (yin1999) [inline files/zh-tw/web/javascript/reference/global_objects/object/entries/index.md:10]: “键控属性”这个翻译感觉并不合适。我比较倾向的翻译是“以字符串为键的属性”

File: files/zh-tw/web/javascript/reference/global_objects/object/entries/index.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+```
+
+### 參數
+
+- `obj`
+  - : 一個物件。
+
+### 回傳值
+
+給定之物件自己的可枚舉字串鍵控屬性的鍵值對的陣列。每個鍵值對都是一個包含兩個元素的陣列：第一個元素是屬性鍵（總是字串），第二個元素是屬性值。
```
Comment (yin1999) [inline files/zh-tw/web/javascript/reference/global_objects/object/entries/index.md:27]: 同

File: files/zh-tw/web/javascript/reference/global_objects/object/entries/index.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+## 描述
+
+`Object.entries()` 傳回一個陣列，其元素是與直接在 `object` 上找到的可枚舉字串鍵控屬性的鍵值對相對應的陣列。這與使用 {{jsxref("Statements/for...in", "for...in")}} 迴圈進行迭代相同，只不過 `for...in` 迴圈也會列舉原型鏈中的屬性。`Object.entries()` 傳回的陣列的順序與 {{jsxref("Statements/for...in", "for...in")}} 迴圈給的順序相同。
```
Comment (yin1999) [inline files/zh-tw/web/javascript/reference/global_objects/object/entries/index.md:31]: 同

File: files/zh-tw/web/javascript/reference/global_objects/object/entries/index.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+        return this.foo;
+      },
+    },
+  },
+);
+myObj.foo = "bar";
+console.log(Object.entries(myObj)); // [ ['foo', 'bar'] ]
+```
+
+### 在 Primitive 上使用 Object.entries()
```
Comment (yin1999) [inline files/zh-tw/web/javascript/reference/global_objects/object/entries/index.md:64]: https://zh.wikipedia.org/zh-tw/%E5%8E%9F%E5%A7%8B%E5%9E%8B%E5%88%A5

```suggestion
### 在原始值上使用 Object.entries()
```

File: files/zh-tw/web/javascript/reference/global_objects/object/entries/index.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+非物件參數會被[強制轉換為物件](/zh-TW/docs/Web/JavaScript/Reference/Global_Objects/Object#object_coercion)，而 [`undefined`](/zh-TW/docs/Web/JavaScript/Reference/Global_Objects/undefined) 和 [`null`](/zh-TW/docs/Web/JavaScript/Reference/Operators/null) 不能被強制轉換為物件，且會先被拋出 {{jsxref("TypeError")}}。僅字串可擁有自己的可枚舉屬性，其它所有的 [primitive](/zh-TW/docs/Glossary/Primitive) 都會變成空陣列。
```
Comment (yin1999) [inline files/zh-tw/web/javascript/reference/global_objects/object/entries/index.md:66]:
```suggestion
非物件參數會被[強制轉換為物件](/zh-TW/docs/Web/JavaScript/Reference/Global_Objects/Object#object_coercion)，而 [`undefined`](/zh-TW/docs/Web/JavaScript/Reference/Global_Objects/undefined) 和 [`null`](/zh-TW/docs/Web/JavaScript/Reference/Operators/null) 不能被強制轉換為物件，且會先拋出 {{jsxref("TypeError")}}。僅字串可擁有自己的可枚舉屬性，其它所有的[原始值](/zh-TW/docs/Glossary/Primitive)都會變成空陣列。
```

File: files/zh-tw/web/javascript/reference/global_objects/object/entries/index.md [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+
+```js
+// 字串具有作為可枚舉自身屬性的索引
+console.log(Object.entries("foo")); // [ ['0', 'f'], ['1', 'o'], ['2', 'o'] ]
+
+// 除 undefined 和 null 之外的其它primitive沒有自己的屬性
```
Comment (yin1999) [inline files/zh-tw/web/javascript/reference/global_objects/object/entries/index.md:72]:
```suggestion
// 除 undefined 和 null 之外的其它原始型別沒有自己的屬性
```

File: files/zh-tw/web/javascript/reference/global_objects/object/entries/index.md [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+console.log(Object.entries(100)); // []
+```
+
+### 將 Object 轉換為 Map
+
+{{jsxref("Map/Map", "Map()")}} 建構函數接受可迭代的 `entries`。你可以藉由 `Object.entries` 輕鬆地將 {{jsxref("Object")}} 轉換為 {{jsxref("Map")}}：
```
Comment (yin1999) [inline files/zh-tw/web/javascript/reference/global_objects/object/entries/index.md:78]: 印象中“建構函數”是翻译成“建構子”的，你可以查阅一下其他的文档

https://terms.naer.edu.tw/search/?csrfmiddlewaretoken=ewtFJxj0q380vuNdKSCH8SnfYk5EPhNl9ylJIpmTkTcZqzLiFPuMZLqcXl8xNbFk&match_type=phrase&query_op=&query_field=title&query_term=constructor&filter_term=&filter_field=&filter_op=&filter_bool=&filter_term=&filter_field=&filter_op=&filter_bool=&filter_term=&filter_field=&filter_op=&filter_bool=&filter_term=&filter_field=&filter_op=&filter_bool=

**Review by yin1999 — APPROVED** [review]

**Pull request merged**